### PR TITLE
vz: qemuPacketConn: Disallow partial reads

### PR DESCRIPTION
**Pull request by tamird**
See commit message.

### CODE REVIEW

**Comment by tamird** [comment]
The current code before this change may result in corruption of network data.

File: pkg/vz/network_darwin.go [hunk 1]
```diff
@@ -98,14 +98,7 @@ func (c *qemuPacketConn) Read(b []byte) (n int, err error) {
...
 	}
-
-	reader := io.LimitReader(c.Conn, int64(size))
-	_, err = reader.Read(b)
-	if err != nil {
-		// Likely connection closed by peer.
-		return 0, err
-	}
-	return int(size), nil
+	return io.ReadFull(c.Conn, b[:size])
```
Comment (nirs) [inline pkg/vz/network_darwin.go:101]: We indeed need to do a full read here, so this change looks correct. The current code may do a short read and return size without checking the amount of data read. This will return a packet with partial data and random junk at end of the buffer left from a previous read.

**Review by nirs — COMMENTED** [review]
Thanks for the PR!

Can you split the actual fix to another PR, or split the unrelated changes to other PRs?

**Comment by tamird** [comment]
@nirs removed the other changes.

File: Makefile [hunk 2]
```diff
@@ -229,8 +232,6 @@ ifneq ($(GOOS),windows) #
 else
 	@rm -rf _output/bin/limactl
 endif
-	# The hostagent must be compiled with CGO_ENABLED=1 so that net.LookupIP() in the DNS server
-	# calls the native resolver library and not the simplistic version in the Go library.
```
[the remark this answers is not in the document]
Comment (tamird) [reply Makefile]: https://github.com/lima-vm/lima/pull/2921

File: pkg/vz/network_darwin.go [hunk 3]
```diff
@@ -27,7 +28,7 @@ func PassFDToUnix(unixSock string) (*os.File, error) {
 	if err != nil {
 		return nil, err
 	}
-	err = fd.Put(unixConn.(*net.UnixConn), server)
```
[the remark this answers is not in the document]
Comment (tamird) [reply pkg/vz/network_darwin.go]: Moved this to https://github.com/lima-vm/lima/pull/2922.

File: pkg/vz/network_darwin.go [hunk 4]
```diff
@@ -116,13 +109,12 @@ func (c *qemuPacketConn) Write(b []byte) (int, error) {
...
-	for start < size {
-		nw, err := c.Conn.Write(b[start:])
+	for len(b) != 0 {
+		n, err := c.Conn.Write(b)
 		if err != nil {
 			return 0, err
 		}
-		start += nw
+		b = b[n:]
 	}
```
Comment (nirs) [inline pkg/vz/network_darwin.go:118]: The write change is not related to the fix - please split it to another PR. We don't want to mix refactoring with bug fixes.

Comment (tamird) [reply pkg/vz/network_darwin.go]: https://github.com/lima-vm/lima/pull/2923

**Review by AkihiroSuda — APPROVED** [review]
Thanks
